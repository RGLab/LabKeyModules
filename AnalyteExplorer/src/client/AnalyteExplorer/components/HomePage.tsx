import React from "react";

interface ArrowTextProps {
  text: string;
}

// same arrow as entrance page
export const ArrowText: React.FC<ArrowTextProps> = ({ text }) => {
  return (
    <a
      href="https://immunespace.org/project/home/begin.view?tab=GeneExpression"
      className="history-block__cta ae-home-arrow-text"
      title="Data Processing Link">
      <span className="history-block__link-text">{`${text}`}</span>
    </a>
  );
};

const HomePage: React.FC = () => {
  return (
    <section className="ae-home-content">
      <div className="ae-home-title">
        <h1>Analyte Explorer</h1>
        <h3>A search tool to visualize study data within ImmuneSpace</h3>
      </div>

      <div className="ae-home-tutorial-instructions">
        <div>
          <img
            src="/AnalyteExplorer/assets/images/tutorial1.png"
            alt="how to use first step"
            className="ae-home__tutorial-icon"
          />
          <span>
            Search our database for analytes related to HIPC study data
          </span>
        </div>
        <div>
          <img
            src="/AnalyteExplorer/assets/images/tutorial2.png"
            alt="how to use second step"
            className="ae-home__tutorial-icon"
          />
          <span>Plot the log2 transform to visualize expression patterns</span>
        </div>
        <div>
          <img
            src="/AnalyteExplorer/assets/images/tutorial3.png"
            alt="how to use third step"
            className="ae-home__tutorial-icon"
          />
          <span>Download your search results and analyze further</span>
        </div>
      </div>
      <div className="ae-home-data-processing">
        <p>
          Analyte Explorer curruently supports gene expression data for genes
          and blood transcription modules (BTMs). For more information about our
          data processing methods for this module visit our data processing
          page.
        </p>

        <ArrowText text="Data Processing" />
      </div>
    </section>
  );
};

export default HomePage;
