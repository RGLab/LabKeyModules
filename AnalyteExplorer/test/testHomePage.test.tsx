/**
 * @jest-environment jsdom
 */

import React from "react";
import { shallow, render, mount } from "enzyme";
import HomePage, {
  ArrowText,
} from "../src/client/AnalyteExplorer/components/HomePage";

describe("<HomePage />", () => {
  it("shallow renders <HomePage />", () => {
    const wrapper = shallow(<HomePage />);
    expect(wrapper.find(".ae-home-content")).toBeDefined();
    expect(wrapper.find(".ae-home-content")).toHaveLength(1);
    expect(wrapper.find(".ae-home__tutorial-icon")).toHaveLength(3);
    expect(wrapper.find(".ae-home-tutorial-instructions")).toHaveLength(1);
    expect(wrapper.find(".ae-home-data-processing")).toHaveLength(1);

    //expect(wrapper.find(".ae-home-arrow-text")).toHaveLength(1);
  });

  it("shallow renders <ArrowText />", () => {
    const text = "Hello";
    const wrapper = render(<ArrowText text={text} />);
    //expect(wrapper.find("a")).toHaveLength(1); // can't find the a tag for some reason
    expect(wrapper.find(".history-block__link-text")).toHaveLength(1);
    expect(wrapper.find(".history-block__link-text").text()).toEqual(text);
  });

  it("full rendering ArrowText", () => {
    const text = "Hello";
    const wrapper = mount(<ArrowText text={text} />);
    expect(wrapper.find(".ae-home-arrow-text")).toHaveLength(1);
    //expect(wrapper.find(".history-block__link-text::after")).toHaveLength(1);
  });
});
