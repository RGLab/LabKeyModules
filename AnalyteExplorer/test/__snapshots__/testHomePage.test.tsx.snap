// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HomePage /> HomePage snapshot test 1`] = `
<section
  className="ae-home-content"
>
  <div
    className="ae-home-title"
  >
    <h1>
      Analyte Explorer
    </h1>
    <h3>
      A search tool to visualize study data within ImmuneSpace
    </h3>
  </div>
  <div
    className="ae-home-tutorial-instructions"
  >
    <div>
      <img
        alt="how to use first step"
        className="ae-home__tutorial-icon"
        src="/AnalyteExplorer/assets/images/tutorial1.png"
      />
      <span>
        Search our database for analytes related to HIPC study data
      </span>
    </div>
    <div>
      <img
        alt="how to use second step"
        className="ae-home__tutorial-icon"
        src="/AnalyteExplorer/assets/images/tutorial2.png"
      />
      <span>
        Plot the log2 fold change of a gene or gene set over time by cohort
      </span>
    </div>
    <div>
      <img
        alt="how to use third step"
        className="ae-home__tutorial-icon"
        src="/AnalyteExplorer/assets/images/tutorial3.png"
      />
      <span>
        Download your search results and analyze further
      </span>
    </div>
  </div>
  <div
    className="ae-home-data-processing"
  >
    <p>
      Analyte Explorer currently supports gene expression data for genes, and blood transcription modules (BTMs) published by 
      <a
        href="https://www.nature.com/articles/ni.2789"
      >
        Li et al. 
      </a>
       Gene metadata is pulled from 
      <a
        href="https://mygene.info/"
      >
        mygene.info
      </a>
      . For details about our data processing methods, visit our
      <a
        href="/project/home/begin.view?tab=GeneExpression"
      >
         Data Processing page 
      </a>
       and
      <a
        href="https://github.com/RGLab/AnalyteExplorer"
      >
         GitHub repository
      </a>
      .
    </p>
  </div>
</section>
`;
